##### 'A_구매금액', '기타CV_S', '식료품_NET','의류_NET', '생활_NET', 'RFM등급', '휴일_CV_S', '편의품_CV_S','선매품CV_S','객단가증감률','B_구매금액','C_구매금액'

In [2]:
import pandas as pd
import numpy as np

df = pd.read_csv('./dataset/라벨붙임.csv',encoding='cp949')
df = df.drop('Unnamed: 0',axis=1)
df

고객번호    A_구매금액    기타CV_S   식료품_NET    의류_NET    생활_NET     RFM등급  \
0          1  1.367753 -0.840909 -0.144455  0.238970 -3.296500 -1.739425   
1          2  1.776146  0.201445 -0.144455  1.282634 -0.051626 -0.792279   
2          3 -0.496782 -1.569289 -0.144455  0.238970 -0.051626 -0.034562   
3          4 -0.170812 -0.657451 -0.144455  0.238970 -0.862844  1.291443   
4         10 -0.400193 -0.190214 -0.144455  0.238970 -0.051626 -0.981708   
...      ...       ...       ...       ...       ...       ...       ...   
10743  19325 -0.207098  0.126255 -0.144455  0.238970 -0.862844  0.723155   
10744  19327 -0.498162  2.347770 -0.144455  1.282634 -0.051626 -0.413420   
10745  19332 -0.493328  2.347770  0.854647  1.282634 -0.051626  0.533726   
10746  19333 -0.438943 -0.713786 -0.144455 -0.804693 -0.862844  1.291443   
10747  19334 -0.488963  0.055907 -1.143558  1.282634 -0.051626 -1.360567   

        휴일_CV_S  편의품_CV_S   선매품CV_S    객단가증감률    B_구매금액    C_구매금액  \
0     -0.015794  0.382423 -0.803469 -0.912943 -0.214188 -0.135190   
1     -0.488643  0.298866 -0.848143 -0.912943 -0.234609 -0.205711   
2      0.466305  0.071407  2.538651 -0.203346 -0.242069  0.018349   
3      0.288886 -0.329763 -0.532511 -0.912943 -0.181465 -0.146886   
4     -1.060247 -1.245738 -0.327869 -0.203346 -0.145213  0.632941   
...         ...       ...       ...       ...       ...       ...   
10743  0.657320  1.350886  0.234927  0.506252 -0.222418 -0.205096   
10744 -0.014097  0.781941  2.607661 -0.912943 -0.239206  0.324242   
10745  0.430470  0.248380  1.333528  1.215849 -0.234873  0.748546   
10746 -0.587946  0.875390  1.377105 -0.912943 -0.202690 -0.203839   
10747  0.129032  1.695419  0.715794 -2.332137  0.178407 -0.148936   

       kmeans_label  
0                 2  
1                 2  
2                 0  
3                 1  
4                 1  
...             ...  
10743             2  
10744             0  
10745             0  
10746             0  
10747             2  

[10748 rows x 14 columns]

In [7]:
df1 = df[df['kmeans_label']==0]
df2 = df[df['kmeans_label']==1]
df3 = df[df['kmeans_label']==2]
df1 = df1[['고객번호']]
df2 = df2[['고객번호']]
df3 = df3[['고객번호']]

In [9]:
dfa=pd.read_csv('./dataset/데이터셋0326.csv',encoding='cp949')
dfa = dfa[['고객번호','A_구매금액','B_구매금액', 'C_구매금액', 'D_구매금액']]

In [11]:
df11 = pd.merge(df1,dfa,on='고객번호')
df21 = pd.merge(df2,dfa,on='고객번호')
df31 = pd.merge(df3,dfa,on='고객번호')

In [81]:
print(f'1군집: {df11.고객번호.count()}   2군집: {df21.고객번호.count()}   3군집: {df31.고객번호.count()}')

1군집: 2571   2군집: 5471   3군집: 2706


In [14]:
display(df11.describe())
display(df21.describe())
display(df31.describe())

고객번호        A_구매금액        B_구매금액        C_구매금액        D_구매금액
count   2571.000000  2.571000e+03  2.571000e+03  2.571000e+03  2.571000e+03
mean   10876.817192  2.131768e+06  4.101321e+06  7.237498e+06  3.295029e+04
std     5502.473438  4.951155e+06  2.100396e+07  2.701257e+07  3.105348e+05
min        3.000000  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00
25%     6265.000000  1.766070e+05  1.433245e+05  9.709500e+04  0.000000e+00
50%    11605.000000  6.553370e+05  8.100530e+05  6.258333e+06  0.000000e+00
75%    15597.000000  1.897962e+06  6.679140e+06  9.728095e+06  0.000000e+00
max    19333.000000  8.765803e+07  9.632265e+08  9.726146e+08  6.114226e+06

고객번호        A_구매금액        B_구매금액        C_구매금액        D_구매금액
count   5471.000000  5.471000e+03  5.471000e+03  5.471000e+03  5.471000e+03
mean    9651.027052  1.125769e+07  8.740621e+06  1.812280e+06  1.947802e+04
std     5385.195067  1.779444e+07  3.326530e+07  4.069610e+06  1.553906e+05
min        4.000000  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00
25%     5033.500000  4.079825e+05  1.137508e+06  1.514000e+04  0.000000e+00
50%     9725.000000  2.362477e+06  8.173210e+06  7.422700e+04  0.000000e+00
75%    14236.500000  1.569750e+07  1.270308e+07  7.767625e+05  0.000000e+00
max    19322.000000  1.568309e+08  2.212347e+09  4.420104e+07  4.051999e+06

고객번호        A_구매금액        B_구매금액        C_구매금액        D_구매금액
count   2706.000000  2.706000e+03  2.706000e+03  2.706000e+03  2.706000e+03
mean    8177.280488  6.331553e+07  4.194489e+06  9.833506e+05  3.012150e+04
std     5464.869808  6.987791e+07  1.296643e+07  6.299787e+06  1.916689e+05
min        1.000000  0.000000e+00  0.000000e+00  0.000000e+00  0.000000e+00
25%     3375.250000  2.033828e+07  1.965168e+05  1.007250e+04  0.000000e+00
50%     7662.000000  5.012966e+07  9.749750e+05  6.396300e+04  0.000000e+00
75%    12665.250000  8.313607e+07  4.089048e+06  3.655450e+05  0.000000e+00
max    19334.000000  1.020549e+09  4.478369e+08  2.856143e+08  4.743887e+06

## 제휴사별 인사이트
1군집: 2571   2군집: 5471      3군집: 2706
- 1군집
    - A :2,131,768	B:4,101,321	C:7,237,498
    - C의 평균구매금액이 큼
- 2군집
    - A: 11,257,690	B:8,740,621	C:1,812,280
    - A와B의 평균구매금액이 큼
- 3군집
    - A:63,315,530	B:4,194,489	C:983,350.6
    - A의 평균구매금액이 매우 큼


In [32]:
dfb1 =pd.read_csv('./dataset/MainCategory/선매품금액횟수_원본.csv',encoding='cp949')
dfb1 = dfb1[dfb1.columns[1:9]]
dfb2 =pd.read_csv('./dataset/MainCategory/편의품.csv')
dfb2 = dfb2[dfb2.columns[1:9]]
dfb3 =pd.read_csv('./dataset/MainCategory/전문품금액횟수_원본.csv',encoding='cp949')
dfb3 = dfb3[dfb3.columns[1:9]]
dfb4 =pd.read_csv('./dataset/MainCategory/기타금액횟수_원본.csv',encoding='cp949')
dfb4 = dfb4[dfb4.columns[1:9]]

In [35]:
#편의품
df121 = pd.merge(df1,dfb2,on='고객번호')
df221 = pd.merge(df2,dfb2,on='고객번호')
df321 = pd.merge(df3,dfb2,on='고객번호')
#선매품
df122 = pd.merge(df1,dfb1,on='고객번호')
df222 = pd.merge(df2,dfb1,on='고객번호')
df322 = pd.merge(df3,dfb1,on='고객번호')
#전문품
df123 = pd.merge(df1,dfb3,on='고객번호')
df223 = pd.merge(df2,dfb3,on='고객번호')
df323 = pd.merge(df3,dfb3,on='고객번호')
#기타
df124 = pd.merge(df1,dfb4,on='고객번호')
df224 = pd.merge(df2,dfb4,on='고객번호')
df324 = pd.merge(df3,dfb4,on='고객번호')

In [47]:
#편
print(df121.describe().iloc[1,-1] - df121.describe().iloc[1,1])  
print(df221.describe().iloc[1,-1] - df221.describe().iloc[1,1]) 
print(df321.describe().iloc[1,-1] - df321.describe().iloc[1,1])  

-181163.33294437965
140232.3388777189
-331345.64671101235


In [48]:
#선
print(df122.describe().iloc[1,-1] - df122.describe().iloc[1,1])  
print(df222.describe().iloc[1,-1] - df222.describe().iloc[1,1]) 
print(df322.describe().iloc[1,-1] - df322.describe().iloc[1,1])

-153857.1968105795
-231166.03089014813
-2523125.240946046


In [49]:
#전
print(df123.describe().iloc[1,-1] - df123.describe().iloc[1,1])  
print(df223.describe().iloc[1,-1] - df223.describe().iloc[1,1]) 
print(df323.describe().iloc[1,-1] - df323.describe().iloc[1,1])

-6210.759626604433
-14505.616706269415
-429326.01589061343


In [50]:
#기
print(df124.describe().iloc[1,-1] - df124.describe().iloc[1,1])  
print(df224.describe().iloc[1,-1] - df224.describe().iloc[1,1]) 
print(df324.describe().iloc[1,-1] - df324.describe().iloc[1,1])

9253.911707506802
33871.56205446903
-3743.1577974870743


### 대분류 7-1 증감
1군집: 2571      2군집: 5471      3군집: 2706
- 1군집: 편의품 감소 -181163.33   선매품 감소 -153857.19  전문품 감소  -6210.75  기타 증가 9253.91
- 2군집: 편의품 증가 140232.33   선매품 감소 -231166.03  전문품 감소  -14505.61   기타 증가 33871.56
- 3군집: 편의품 감소 -331345.64   선매품 감소 -2523125.24  전문품 감소  -429326.01   기타 감소 -3743.15

In [83]:
print (f'1군집 평균구매금액 편의품: {a1}  선매품: {b1}  전문품: {c1}  기타: {d1}')
print (f'2군집 평균구매금액 편의품: {a2}  선매품: {b2}  전문품: {c2}  기타: {d2}')
print (f'3군집 평균구매금액 편의품: {a3}  선매품: {b3}  전문품: {c3}  기타: {d3}')

1군집 평균구매금액 편의품: 1451001.9  선매품: 277655.22  전문품: 15013.61  기타: 28126.43
2군집 평균구매금액 편의품: 1375832.99  선매품: 1242217.99  전문품: 121188.8  기타: 115937.48
3군집 평균구매금액 편의품: 1583368.48  선매품: 5940280.96  전문품: 1307939.34  기타: 186963.12


In [82]:
#편
a1 = round((df121.describe().iloc[1,1:].sum()/7),2)
a2 = round((df221.describe().iloc[1,1:].sum()/7),2)
a3 = round((df321.describe().iloc[1,1:].sum()/7),2)

b1 = round((df122.describe().iloc[1,1:].sum()/7),2)
b2 = round((df222.describe().iloc[1,1:].sum()/7),2)
b3 = round((df322.describe().iloc[1,1:].sum()/7),2)

c1 = round((df123.describe().iloc[1,1:].sum()/7),2)
c2 = round((df223.describe().iloc[1,1:].sum()/7),2)
c3 = round((df323.describe().iloc[1,1:].sum()/7),2)

d1 = round((df124.describe().iloc[1,1:].sum()/7),2)
d2 = round((df224.describe().iloc[1,1:].sum()/7),2)
d3 = round((df324.describe().iloc[1,1:].sum()/7),2)

### 대분류 평균구매금액
1군집: 2571      2군집: 5471      3군집: 2706
- 1군집: 선매품/전문품/기타 구매금액 제일 작음
- 2군집: 편의품/선매품의 구매 비중이 큼
- 3군집: 편의품 대비 선매품/전문품/기타 구매금액이 큼

In [74]:
dfc1 =pd.read_csv('./dataset/MiddleCategory/식료품_원본.csv',encoding='cp949')
dfc1 = dfc1[dfc1.columns[1:9]]
dfc2 =pd.read_csv('./dataset/MiddleCategory/의류잡화_원본.csv',encoding='cp949')
dfc2 = dfc2[dfc2.columns[1:9]]
dfc3 =pd.read_csv('./dataset/MiddleCategory/생활잡화_원본.csv',encoding='cp949')
dfc3 = dfc3[dfc3.columns[1:9]]

In [76]:
#식료품
df131 = pd.merge(df1,dfc1,on='고객번호')
df231 = pd.merge(df2,dfc1,on='고객번호')
df331 = pd.merge(df3,dfc1,on='고객번호')
#의류품
df132 = pd.merge(df1,dfc2,on='고객번호')
df232 = pd.merge(df2,dfc2,on='고객번호')
df332 = pd.merge(df3,dfc2,on='고객번호')
#생활품
df133 = pd.merge(df1,dfc3,on='고객번호')
df233 = pd.merge(df2,dfc3,on='고객번호')
df333 = pd.merge(df3,dfc3,on='고객번호')

In [77]:
#식료
print(df131.describe().iloc[1,-1] - df131.describe().iloc[1,1])  
print(df231.describe().iloc[1,-1] - df231.describe().iloc[1,1]) 
print(df331.describe().iloc[1,-1] - df331.describe().iloc[1,1])  

-89696.58304161811
102152.0007311278
-159047.79600886907


In [78]:
#의류
print(df132.describe().iloc[1,-1] - df132.describe().iloc[1,1])  
print(df232.describe().iloc[1,-1] - df232.describe().iloc[1,1]) 
print(df332.describe().iloc[1,-1] - df332.describe().iloc[1,1])  

-106871.22520420069
-207450.98117346002
-2138787.620842572


In [79]:
#생활
print(df133.describe().iloc[1,-1] - df133.describe().iloc[1,1])  
print(df233.describe().iloc[1,-1] - df233.describe().iloc[1,1]) 
print(df333.describe().iloc[1,-1] - df333.describe().iloc[1,1])  

-95745.45390898481
36331.47505026506
-130748.33702882484


### 세분류(식료/의류/생활) 7-1 증감
1군집: 2571      2군집: 5471      3군집: 2706
- 1군집: 식품 평균구매액 감소 -89696.58    의류 평균구매액 감소 -106871.22   생활 평균구매액 감소 -95745.45
    - 의류잡화부분 감소
- 2군집: 식품 평균구매액 증가 102152.00    의류 평균구매액 감소 -207450.98   생활 평균구매액 증가 36331.47
    - 다른 제품 증가 의류부분 감소
- 3군집: 식품 평균구매액 감소 -159047.79   의류 평균구매액 감소 -2138787.62  생활 평균구매액 감소 -130748.33
    - 의류잡화부분 감소,전체적으로 감소


In [86]:
print (f'1군집 평균구매금액 식료: {a11}  의류: {b11}  생활: {c11} ')
print (f'2군집 평균구매금액 식료: {a22}  의류: {b22}  생활: {c22} ')
print (f'3군집 평균구매금액 식료: {a33}  의류: {b33}  생활: {c33} ')

1군집 평균구매금액 식료: 1080668.02  의류: 198127.84  생활: 160676.68 
2군집 평균구매금액 식료: 1007545.74  의류: 841892.9  생활: 200955.77 
3군집 평균구매금액 식료: 1090878.81  의류: 3905929.68  생활: 353371.96 


In [85]:
a11 = round((df131.describe().iloc[1,1:].sum()/7),2)
a22 = round((df231.describe().iloc[1,1:].sum()/7),2)
a33 = round((df331.describe().iloc[1,1:].sum()/7),2)

b11 = round((df132.describe().iloc[1,1:].sum()/7),2)
b22 = round((df232.describe().iloc[1,1:].sum()/7),2)
b33 = round((df332.describe().iloc[1,1:].sum()/7),2)

c11 = round((df133.describe().iloc[1,1:].sum()/7),2)
c22 = round((df233.describe().iloc[1,1:].sum()/7),2)
c33 = round((df333.describe().iloc[1,1:].sum()/7),2)


In [88]:
dfd =pd.read_csv('./dataset/데이터셋0326.csv',encoding='cp949')
dfd = dfd[['고객번호','구매빈도','최근구매일','총구매금액']]
dfd

고객번호  구매빈도  최근구매일      총구매금액
0          1   979     10   84772748
1          2   997      3  101633022
2          3  1098     10    3250398
3          4   823      2   17042247
4          6  1374      2   24771126
...      ...   ...    ...        ...
18861  19333   734      4    4769814
18862  19334  1916     33   12491942
18863  19335  1643      2   16917268
18864  19336   389      7   30681024
18865  19337  1054      3    4984754

[18866 rows x 4 columns]

In [89]:
df141 = pd.merge(df1,dfd,on='고객번호')
df241 = pd.merge(df2,dfd,on='고객번호')
df341 = pd.merge(df3,dfd,on='고객번호')

In [95]:
df141

고객번호  구매빈도  최근구매일     총구매금액
0         3  1098     10   3250398
1        12   427      2   6607296
2        28  1276      6   8852069
3        33  2115      4  10671169
4        43   979      6   7383116
...     ...   ...    ...       ...
2566  19309  2084      4  11628922
2567  19320  1384      2  18692216
2568  19327  1336     11   7579908
2569  19332  2613      2  13895294
2570  19333   734      4   4769814

[2571 rows x 4 columns]

In [91]:
display(df141.describe())
display(df241.describe())
display(df341.describe())

고객번호         구매빈도        최근구매일         총구매금액
count   2571.000000  2571.000000  2571.000000  2.571000e+03
mean   10876.817192  1672.003890     3.220537  1.350354e+07
std     5502.473438   707.599362     3.643872  4.389689e+07
min        3.000000    35.000000     2.000000  9.397290e+05
25%     6265.000000  1176.500000     2.000000  8.046138e+06
50%    11605.000000  1594.000000     2.000000  1.049943e+07
75%    15597.000000  2097.500000     3.000000  1.403168e+07
max    19333.000000  9096.000000    65.000000  1.736919e+09

고객번호         구매빈도        최근구매일         총구매금액
count   5471.000000  5471.000000  5471.000000  5.471000e+03
mean    9651.027052  1507.121550     3.824347  2.183007e+07
std     5385.195067   697.840699     3.226326  3.625894e+07
min        4.000000   213.000000     2.000000  4.895440e+05
25%     5033.500000   965.000000     2.000000  1.161148e+07
50%     9725.000000  1418.000000     3.000000  1.599852e+07
75%    14236.500000  1918.000000     4.000000  2.441751e+07
max    19322.000000  6254.000000    67.000000  2.219547e+09

고객번호         구매빈도        최근구매일         총구매금액
count   2706.000000  2706.000000  2706.000000  2.706000e+03
mean    8177.280488   933.133407     7.202882  6.852349e+07
std     5464.869808   511.539249     8.970719  7.033101e+07
min        1.000000    22.000000     2.000000  4.335020e+05
25%     3375.250000   595.250000     3.000000  2.550512e+07
50%     7662.000000   782.000000     4.000000  5.363976e+07
75%    12665.250000  1118.000000     8.000000  8.651449e+07
max    19334.000000  5822.000000    90.000000  1.021644e+09

In [94]:
df341.describe().iloc[1,:]

고객번호     8.177280e+03
구매빈도     9.331334e+02
최근구매일    7.202882e+00
총구매금액    6.852349e+07
Name: mean, dtype: float64

### RFM
1군집: 2571      2군집: 5471      3군집: 2706
- 1군집: 평균 총구매액: 13,503,540 최근구매일: 3.22 구매빈도: 1672
- 2군집: 평균 총구매액: 21,830,070 최근구매일: 3.82 구매빈도: 1507
- 3군집: 평균 총구매액: 68,523,490 최근구매일: 7.2 구매빈도: 933

In [96]:
dfe =pd.read_csv('./dataset/휴일데이터.csv',encoding='cp949')
dfe

고객번호         1Q         2Q         3Q         4Q         5Q  \
0          1  1811389.0  3512459.0  1884171.0  5953286.0  4646550.0   
1          2  4041389.0  3788428.0  3371187.0  2881929.0  8402930.0   
2          3    59132.0    66277.0    87275.0   123290.0    75275.0   
3          4   196644.0   876922.0   920872.0   215075.0   241957.0   
4          5   705305.0  1192831.0   629121.0    56136.0  1263538.0   
...      ...        ...        ...        ...        ...        ...   
19361  19379        NaN        NaN   116031.0   273679.0   195855.0   
19362  19380        NaN        NaN        NaN        NaN   178501.0   
19363  19381        NaN        NaN        NaN        NaN   481405.0   
19364  19382        NaN        NaN        NaN        NaN   267723.0   
19365  19383        NaN        NaN        NaN        NaN        NaN   

              6Q         7Q         8Q  
0      3057190.0  1274454.0   302459.0  
1      4458696.0  4648400.0  4175131.0  
2        10755.0   199681.0    42182.0  
3       448275.0   577373.0   589834.0  
4       496183.0        NaN   788232.0  
...          ...        ...        ...  
19361   350913.0   355143.0   362337.0  
19362   149420.0   253984.0   815885.0  
19363   472162.0   621006.0   321228.0  
19364   236677.0   370585.0   345102.0  
19365    19302.0   626270.0   658371.0  

[19366 rows x 9 columns]

In [97]:
df151 = pd.merge(df1,dfe,on='고객번호')
df251 = pd.merge(df2,dfe,on='고객번호')
df351 = pd.merge(df3,dfe,on='고객번호')

In [100]:
df151.describe().iloc[1,-2] 

482835.4322631167

In [101]:
df151.describe()

고객번호            1Q            2Q            3Q            4Q  \
count   2570.000000  2.559000e+03  2.564000e+03  2.559000e+03  2.563000e+03   
mean   10877.293774  6.734834e+05  5.887006e+05  6.175307e+05  5.065649e+05   
std     5503.491197  4.046952e+06  2.353756e+06  2.077607e+06  1.669641e+06   
min        3.000000  4.231000e+03  3.605000e+03  6.530000e+03  1.419000e+03   
25%     6262.500000  2.498580e+05  2.479940e+05  2.814065e+05  2.153440e+05   
50%    11606.000000  3.959320e+05  3.951375e+05  4.350340e+05  3.291040e+05   
75%    15601.000000  6.449635e+05  6.098655e+05  6.855910e+05  5.385680e+05   
max    19333.000000  1.814100e+08  1.051863e+08  8.415762e+07  7.428921e+07   

                 5Q            6Q            7Q            8Q  
count  2.565000e+03  2.562000e+03  2.554000e+03  2.546000e+03  
mean   5.469265e+05  5.347088e+05  4.828354e+05  3.208053e+05  
std    1.376641e+06  1.989907e+06  9.794636e+05  7.414986e+05  
min    5.077000e+03  2.145000e+03  1.791000e+03  3.470000e+03  
25%    2.446110e+05  2.374632e+05  2.212470e+05  1.432030e+05  
50%    3.910110e+05  3.769375e+05  3.546265e+05  2.333180e+05  
75%    6.133480e+05  5.690668e+05  5.602610e+05  3.772605e+05  
max    6.247963e+07  9.314063e+07  3.399905e+07  3.127949e+07

In [102]:
# 휴일 7-1 증감
print(df151.describe().iloc[1,-2] - df151.describe().iloc[1,1])
print(df251.describe().iloc[1,-2] - df251.describe().iloc[1,1])
print(df351.describe().iloc[1,-2] - df351.describe().iloc[1,1])

-190647.976099525
-161381.61556842772
-1306824.426812097


In [105]:
print(round((df151.describe().iloc[1,1:-1].sum()/7),2))
print( round((df251.describe().iloc[1,1:-1].sum()/7),2))
print( round((df351.describe().iloc[1,1:-1].sum()/7),2))

564392.9
1118764.94
3708832.3
